### ISSUE TITLE
[BUG 4.2.2] New performance issue with "include count"

### ISSUE THREAD

**Issue by tvarsis** [dreamfactorysoftware/dreamfactory#369]
For larger tables, when using the option "include count" the query performance slows down extremely much. Did not have this issue on earlier 4.0.x version. For example, do a filter query on a large table with 40m rows in it that should returns 300k rows, and adding a limit = 10 takes about 1 sec to finish. Adding the include count to that one which then also returns the 300k in the count prop, increases that query to about 25 sec. Running the same count() with the filter directly in the db takes about 1 sec.

Version of DF: 4.2.2
PostgreSQL 12
Using official docker images from dockerhub running on Ubuntu 18.04

**Comment by wjgilmore** [comment]
Hi @tvarsis 

Thank you for this report. Did your PostgreSQL version happen to change between versions?

Jason

**Comment by tvarsis** [comment]
Hi Jason,

Only minor bump in that case. Probably went from 12.2 to 12.3. But I think I ran 12.3 on 4.0.x as well and I think I did not see this issue then.

**Comment by wjgilmore** [comment]
OK thank you we will look into this. Would love to learn more about what you're doing with DreamFactory! jason.gilmore@dreamfactory.com

**Comment by tvarsis** [comment]
We managed to identify the issue which was actually due to a FILTER that we had doing lower(column) on that tabled which added the overhead to the query which gave us the different results from within DF vs directly in the DB. We found that out by inspecting the actual query that DF generated at runtime. Removing that FILTER function solved this issue and DF are now having the same performance as query directly in the DB. Closing this issue and apologies for any inconvenience!
